### ISSUE TITLE
[studio] Hide Live Items button in My Recent Activity widget does not work after click it

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#5601]
#### Crafter CMS Version
{{Crafter CMS version (copy paste from Help > About).}}
3.1.24-SNAPSHOT-e7312d

#### Date of Build
May 20, 2022

#### Describe the bug

My Recent Activity widget does not show any element when click Hide Live Items, when it should keep the edited pages displayed in the panel.
I think the issue is in the API service /get-user-activities.json that is returning 0 results.
#### To Reproduce

1. In preview, publish two pages.
2. Edit one of those pages and do not publish it
3. Go to the Dashboard
4. Scroll down to My Recent Activity panel
5. Verify you have one page with status "edited" and the other one with status "live".
6. Click Hide Live Items.
7. Panel shows 0 results, but it should keep the edited page displayed.

#### Logs

```
[DEBUG] 2022-05-20T10:30:55,295 [http-nio-8080-exec-6] [AuditDAO.selectUserFeedEntriesHideLive] | ==>  Preparing: select act.* from ( select cs1.* from audit cs1 inner join (select MAX(operation_timestamp) as mdate, primary_target_value from audit a, site s1 where a.site_id = s1.id and actor_id = ? and s1.site_id = ? and s1.deleted = 0 and operation in ( ? , ? , ? , ? ) and primary_target_type = ? group by primary_target_value) as cs2 on (cs1.primary_target_value = cs2.primary_target_value and cs1.operation_timestamp = cs2.mdate), site s2 where cs1.site_id = s2.id and cs1.actor_id = ? and s2.site_id = ? and s2.deleted = 0 and cs1.operation in ( ? , ? , ? , ? ) and primary_target_type = ? ) act inner join item_state os on act.site_id=os.site and act.primary_target_value=os.path, site s3 where act.site_id = s3.id and act.actor_id = ? and s3.site_id = ? and s3.deleted = 0 and act.operation in ( ? , ? , ? , ? ) and primary_target_type = ? and os.state NOT IN ( ? , ? , ? ) order by act.operation_timestamp desc limit ?,?  
[DEBUG] 2022-05-20T10:30:55,296 [http-nio-8080-exec-6] [AuditDAO.selectUserFeedEntriesHideLive] | ==> Parameters: admin(String), 500(String), CREATE(String), DELETE(String), UPDATE(String), MOVE(String), Content Item(String), admin(String), 500(String), CREATE(String), DELETE(String), UPDATE(String), MOVE(String), Content Item(String), admin(String), 500(String), CREATE(String), DELETE(String), UPDATE(String), MOVE(String), Content Item(String), EXISTING_UNEDITED_LOCKED(String), EXISTING_UNEDITED_UNLOCKED(String), NEW_DELETED(String), 0(Integer), 100(Integer) 
[DEBUG] 2022-05-20T10:30:55,298 [http-nio-8080-exec-6] [AuditDAO.selectUserFeedEntriesHideLive] | <==      Total: 0 
```

#### Screenshots
{{If applicable, add screenshots to help explain your problem.}}

**Comment by phuongnq** [comment]
The query seems incorrect as we compare `inner join item_state os on act.site_id=os.site`. In this case `site_id` is from `audit` table with data type `bigint(20)` while `item_state` `site` is `varchar(50)`. A possible fix could be using `site_id` from `site` table to do the join.

```
select act.* from ( select cs1.*, cs2.siteId from audit cs1 inner join (select MAX(operation_timestamp) as mdate, primary_target_value, s1.site_id as siteId from audit a, site s1 where a.site_id = s1.id and actor_id = "admin" and s1.site_id = "editorial" and s1.deleted = 0 and operation in ( "CREATE", "DELETE", "UPDATE", "MOVE" ) and primary_target_type = "Content Item" group by primary_target_value) as cs2 on (cs1.primary_target_value = cs2.primary_target_value and cs1.operation_timestamp = cs2.mdate), site s2 where cs1.site_id = s2.id and cs1.actor_id = "admin" and s2.site_id = "editorial" and s2.deleted = 0 and cs1.operation in ( "CREATE", "DELETE", "UPDATE", "MOVE" ) and primary_target_type = "Content Item" ) act inner join item_state os on act.siteId=os.site and act.primary_target_value=os.path, site s3 where act.site_id = s3.id and act.actor_id = "admin" and s3.site_id = "editorial" and s3.deleted = 0 and act.operation in ( "CREATE", "DELETE", "UPDATE", "MOVE" ) and primary_target_type = "Content Item" 
and os.state NOT IN ( "EXISTING_UNEDITED_LOCKED", "EXISTING_UNEDITED_UNLOCKED", "NEW_DELETED" ) order by act.operation_timestamp desc limit 0,100;  
```

**Comment by phuongnq** [comment]
@sumerjabri  In v4, there is no issue as we do the INNER JOIN with `item` table which has `site_id` BIGINT

**Comment by AndyReiter** [comment]
Confirmed fix in 3.1.25-SNAPSHOT.
